src/cli.c

```diff
@@ -53,16 +53,152 @@ void print_help(void)
+                case 1: return "short";
+                case 2: return "short2";
+                case 3: return "short3";
+                case 4: return "short4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_USHORT:
+            switch (vector_size) {
+                case 1: return "ushort";
+                case 2: return "ushort2";
+                case 3: return "ushort3";
+                case 4: return "ushort4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_INT:
+            switch (vector_size) {
+                case 1: return "int";
+                case 2: return "int2";
+                case 3: return "int3";
+                case 4: return "int4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_UINT:
+            switch (vector_size) {
+                case 1: return "uint";
+                case 2: return "uint2";
+                case 3: return "uint3";
+                case 4: return "uint4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_LONG:
+            switch (vector_size) {
+                case 1: return "long";
+                case 2: return "long2";
+                case 3: return "long3";
+                case 4: return "long4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_ULONG:
+            switch (vector_size) {
+                case 1: return "ulong";
+                case 2: return "ulong2";
+                case 3: return "ulong3";
+                case 4: return "ulong4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_HALF:
+            switch (vector_size) {
+                case 1: return "half";
+                case 2: return "half2";
+                case 3: return "half3";
+                case 4: return "half4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_FLOAT:
+            switch (vector_size) {
+                case 1: return "float";
+                case 2: return "float2";
+                case 3: return "float3";
+                case 4: return "float4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_DOUBLE:
+            switch (vector_size) {
+                case 1: return "double";
+                case 2: return "double2";
+                case 3: return "double3";
+                case 4: return "double4";
+                default: break;
+            }
+            break;
+        case SDL_SHADERCROSS_IOVAR_TYPE_UNKNOWN:
+        default: break;
+    }
+
+    SDL_LogWarn(SDL_LOG_CATEGORY_APPLICATION, "Unknown IO variable type: vector_type=%u vector_size=%u", io_var_type, vector_size);
```

Comment:
Running `shadercross` on the `SDL_gpu_examples` spirv shaders, I see this warning:
```
$ shadercross -s SPIRV -d JSON  CustomSampling.frag.spv -o a.json
WARNING: Unknown IO variable type: vector_type=0 vector_size=0

```